src/node.h

```diff
@@ -591,14 +591,14 @@ class AsyncResource {
     AsyncResource(v8::Isolate* isolate,
                   v8::Local<v8::Object> resource,
                   const char* name,
-                  async_uid trigger_id = -1)
+                  async_uid trigger_async_id = -1)
         : isolate_(isolate),
           resource_(isolate, resource),
-          trigger_id_(trigger_id) {
-      if (trigger_id_ == -1)
-        trigger_id_ = AsyncHooksGetTriggerId(isolate);
+          trigger_async_id_(trigger_async_id) {
+      if (trigger_async_id_ == -1)
+        trigger_async_id_ = AsyncHooksGetTriggerAsyncId(isolate);
```

Comment:
Oh, right. It is the `AsyncHooksGetTriggerAsyncId` implementation that is wrong then, it should return `async_uid_fields[kCurrentTriggerId]` not `async_uid_fields[kInitTriggerId]`.

The actual behavior in `AsyncResource` is fine.